components/Navbar.vue

```diff
@@ -145,173 +143,124 @@ import ExploreDropdown from '~/components/navbar/ExploreDropdown.vue'
-    BasicImage,
-    MobileExpandableSection,
-    ProfileDropdown,
-    ExploreDropdown,
-    CreateDropdown,
-    ChainSelectDropdown,
-    StatsDropdown,
-    MobileNavbarProfile,
-    NavbarExploreOptions,
-    ConnectWalletButton,
-    ColorModeButton,
-    MobileLanguageOption,
-  },
-})
-export default class NavbarMenu extends mixins(
-  PrefixMixin,
-  AuthMixin,
-  ExperimentMixin
-) {
-  public showTopNavbar = true
-  public openMobileSearchBar = false
-  private fixedTitleNavAppearDistance = 85
-  private lastScrollPosition = 0
-  private artistName = ''
-  public isBurgerMenuOpened = false
-  private isMobile = window.innerWidth < 1024
+const { $store, $colorMode, $buefy, $nextTick } = useNuxtApp()
+const root = ref<Vue<Record<string, string>>>()
+const showTopNavbar = ref(true)
+const openMobileSearchBar = ref(false)
+const fixedTitleNavAppearDistance = ref(85)
+const lastScrollPosition = ref(0)
+const isBurgerMenuOpened = ref(false)
+const isMobile = ref(window.innerWidth < 1024)
+const { urlPrefix } = usePrefix()
 
-  @Ref('mobilSearchRef') readonly mobilSearchRef
+const mobilSearchRef = ref<{ focusInput: () => void } | null>(null)
 
-  get account() {
-    return this.$store.getters.getAuthAddress
-  }
+const route = useRoute()
 
-  get chain(): string {
-    return this.urlPrefix
-  }
+const account = computed(() => $store.getters.getAuthAddress)
 
-  get tokens() {
-    return ['', getKusamaAssetId(this.urlPrefix)]
-  }
+const isCreateVisible = computed(() => createVisible(urlPrefix.value))
+const isLandingPage = computed(() => route.name === 'index')
 
-  get isCreateVisible(): boolean {
-    return createVisible(this.urlPrefix)
-  }
+const isDarkMode = computed(
+  () =>
+    $colorMode.preference === 'dark' ||
+    document.documentElement.className.includes('dark-mode')
+)
 
-  get currentCollection() {
-    return this.$store.getters['history/getCurrentlyViewedCollection'] || {}
-  }
+const logoSrc = computed(() => (isDarkMode.value ? KodaBetaDark : KodaBeta))
 
-  get currentGalleryItemName() {
-    return this.$store.getters['history/getCurrentlyViewedItem']?.name || ''
-  }
+const showSearchOnNavbar = computed(
+  () => !isLandingPage.value || !showTopNavbar.value || isBurgerMenuOpened.value
+)
 
-  get isLandingPage() {
-    return this.$route.name === 'index'
-  }
+const openWalletConnectModal = (): void => {
+  closeBurgerMenu()
 
-  get isDarkMode() {
-    return (
-      this.$colorMode.preference === 'dark' ||
-      document.documentElement.className.includes('dark-mode')
-    )
-  }
+  $buefy.modal.open({
+    parent: root?.value,
+    ...ConnectWalletModalConfig,
+  } as any as BModalConfig)
+}
 
-  get logoSrc() {
-    return this.isDarkMode ? KodaBetaDark : KodaBeta
-  }
+watch([isBurgerMenuOpened], () => {
+  setBodyScroll(!isBurgerMenuOpened.value)
+})
 
-  get showSearchOnNavbar(): boolean {
-    return !this.isLandingPage || !this.showTopNavbar || this.isBurgerMenuOpened
+const onScroll = () => {
```

Comment:
Also @preschian used some composable for bottom

Reply:
I'm not sure what you mean TBH 👀  Which part or composable api could I reuse

Reply:
> I'm not sure what you mean TBH 👀

Was thinking if something like this would help https://vueuse.org/core/useScroll/

Reply:
After some trying, I found that `useScroll` could not be used in this case.